### PR TITLE
MIGENG-355: CNV Affinity Rules

### PR DESCRIPTION
**Pull request by m-brophy**
https://issues.redhat.com/browse/MIGENG-355

### CODE REVIEW

File: src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java [hunk 1]
```diff
@@ -36,6 +36,8 @@
     private String version;
     private String host_name;
     private boolean cpuAffinityNotNull;
+    private boolean numaNodeAffinityNotNull;
```
Comment (jonathanvila) [inline src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java]: In the MA-OCPv validation rules spreadsheet it is said that numaNodeAffinity is a String field

**Review by jonathanvila — COMMENTED** [review]
Considering that some fields are applicable when the source is MTV , I assume that in other cases these fields will come as NULL therefore defining them as "boolean" primitive type will make system crash or would require a default affecting the business rules fired.
I would suggest to change them to Boolean class.
What do you think ?

File: src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java (hunk 1 above)
Comment (m-brophy) [reply src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java]: I agree, though looking at the validation rules spreadsheet cpu affinity is defined as an array of int or null - will the integration project convert this into Boolean?

Comment (m-brophy) [reply src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java]: just remembered that bit had already been done

Comment (m-brophy) [reply src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java]: for numaNodeAffinity are we just checking that it is not null to raise the flag? There's no mention of requiring a particular value.

Comment (jonathanvila) [reply src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java]: In fact , I don't see any mention to raise a flag. ¿?

Comment (m-brophy) [reply src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java]: The JIRA comments mention a flag foe each of the 3 fields

Comment (jonathanvila) [reply src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java]: VM/Host affinity/anti-affinity - generate Flag if has_vm_affinity_config equals true
CPU affinity                         - generate Flag if cpu_affinity not null
NUMA node affinity              - generate Flag if numa_node_affinity not null

File: src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java [hunk 2]
```diff
@@ -35,7 +35,9 @@
     private String product;
     private String version;
     private String host_name;
-    private boolean cpuAffinityNotNull;
+    private Boolean cpuAffinityNotNull;
```
Comment (jonathanvila) [inline src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java]: If now it's a class, and can be nullable..... would it be more appropiate to name the filed "cpuAffinity" ?

Comment (m-brophy) [reply src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java]: Done

File: src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java (hunk 1 above)
Comment (m-brophy) [reply src/main/java/org/jboss/xavier/analytics/pojo/input/workload/inventory/VMWorkloadInventoryModel.java]: rules altered to match this spec

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java [hunk 3]
```diff
@@ -22,6 +22,8 @@
     public static final String SHARED_DISK_FLAG_NAME = "Shared Disk";
     public static final String CPU_MEMORY_HOTPLUG_FLAG_NAME = "CPU/Memory hotplug";
     public static final String CPU_AFFINITY_FLAG_NAME = "CPU Affinity";
+    public static final String VM_HOST_AFFINITY_CONFIGURED_FLAG_NAME = "VM/Host Affinity Configured";
```
Comment (jonathanvila) [inline src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java:25]: These 2 new Flags need to be created in the flag_assessment_model table

Comment (jonathanvila) [reply src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java:25]: You can create a new SQL Flyway script following the approach in MIGENG-795, once that one is merged

**Review by jonathanvila — APPROVED** [review]
LGTM :+1:

**Pull request merged**